lib/v2/google/search.js

```diff
@@ -0,0 +1,57 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+const renderDescription = (description, images) => art(path.join(__dirname, './templates/description.art'), { description, images });
+
+module.exports = async (ctx) => {
+    const { keyword, language } = ctx.params;
+    const searchParams = new URLSearchParams({
+        q: keyword,
+    });
+    const _url = new URL('https://www.google.com/search');
+    _url.search = searchParams.toString();
+    const url = _url.toString();
+    const key = `google-search:${language}:${url}`;
+    const response = await ctx.cache.tryGet(
+        key,
+        async () =>
+            (
+                await got(url, {
+                    headers: {
+                        'Accept-Language': language,
+                    },
+                })
+            ).data
+    );
```

Comment:
- Use the cache efficiently. Do not cache the whole HTML. Cache the processed output instead.

- Please also note that you're calling [`tryGet()`](https://docs.rsshub.app/joinus/advanced/use-cache#ctxcachetrygetkey-getvaluefunc--maxage--refresh-) without disabling `refesh`. The content will be kept extending its expiration time whenever it's called using the same key.